include/eve/module/core/regular/impl/simd/arm/sve/first_true.hpp

```diff
@@ -9,54 +9,59 @@
-template<relative_conditional_expr C, logical_simd_value L>
-EVE_FORCEINLINE std::optional<std::ptrdiff_t>
-                first_true_(EVE_SUPPORTS(sve_), C c, L m) noexcept
-{
-  if constexpr( C::is_complete && !C::is_inverted ) return std::nullopt;
-  else if constexpr( has_aggregated_abi_v<L> )
-  {
-    if constexpr( !C::is_complete ) m = m && sve_true(c, as(m));
-    auto [lo, hi] = m.slice();
-    auto lo_res   = first_true[ignore_none](lo);
-    auto hi_res   = first_true[ignore_none](hi);
-    if( lo_res ) return lo_res;
-    if( hi_res ) *hi_res += lo.size();
-    return hi_res;
-  }
-  else
+  // There are some explanations
+  // Here: https://lemire.me/blog/2022/12/19/implementing-strlen-using-sve/
+  // Or: https://www.stonybrook.edu/commcms/ookami/support/_docs/5%20-%20Advanced%20SVE.pdf
+  template<callable_options O, typename T, typename N>
+  EVE_FORCEINLINE std::optional<std::ptrdiff_t> first_true_(EVE_REQUIRES(sve_), O const& opts, logical<wide<T, N>> m) noexcept
+    requires sve_abi<abi_t<T, N>>
   {
-    auto c_m = L {sve_true(c, eve::as(m))};
+    using L = logical<wide<T, N>>;
+    using C = rbr::result::fetch_t<condition_key, O>;
+    auto cx = opts[condition_key];
 
-    // We don't need this much but it makes the `no matches` case
-    // faster
-    if( !svptest_any(c_m, m) ) return std::nullopt;
+    if constexpr (C::is_complete && !C::is_inverted) return std::nullopt;
+    else if constexpr (has_aggregated_abi_v<L>)
+    {
+      if constexpr (!C::is_complete) m = m && sve_true(cx, as(m));
 
-    // we need to ignore as false
-    if constexpr( !C::is_complete ) m = m && c_m;
+      auto [lo, hi] = m.slice();
+      auto lo_res   = first_true(lo);
+      auto hi_res   = first_true(hi);
 
-    eve::as_wide_t<eve::element_type_t<L>> first_true_mask =
-        svbrkb_z(sve_true<element_type_t<L>>(), m);
-    return count_true(first_true_mask);
-  }
-}
+      if (lo_res) return lo_res;
+      if (hi_res) *hi_res += lo.size();
 
-template<logical_simd_value L>
-EVE_FORCEINLINE std::optional<std::ptrdiff_t>
-                first_true_(EVE_SUPPORTS(sve_), L m) noexcept
-{
-  return first_true[ignore_none](m);
-}
+      return hi_res;
+    }
+    else
+    {
+      L cx_m;
 
-template<logical_simd_value L>
-EVE_FORCEINLINE std::optional<std::ptrdiff_t>
-                first_true_(EVE_SUPPORTS(sve_), top_bits<L> m) noexcept
-{
-  return first_true(to_logical(m));
-}
+      if constexpr (relative_conditional_expr<C>)
+      {
+        cx_m = L {sve_true(cx, eve::as(m))};
+      }
+      else
+      {
+        cx_m = expand_mask(cx, as<L>{});
+      }
+
+      // We don't need this much but it makes the `no matches` case faster
+      if (!svptest_any(cx_m, m)) return std::nullopt;
+
+      // we need to ignore as false
+      if constexpr (!C::is_complete) m = m && cx_m;
 
+      as_wide_t<element_type_t<L>> first_true_mask = svbrkb_z(sve_true<element_type_t<L>>(), m);
```

Comment:
I have no idea what I was doing here.
Any idea why it's `as_wide_t<element_type_t<L>>` and not L?
cheaper in count true?